### PR TITLE
Make tag field grouping case ignored

### PR DESCRIPTION
**Pull request by wangsongiam**
related issue: https://github.com/gatsbyjs/gatsby/issues/1789

So far I've coded the case insensitivity on tags field, I'll try to figure out adding an option in Graphql.

### CODE REVIEW

File: packages/gatsby/src/schema/build-connection-fields.js [hunk 1]
```diff
@@ -104,7 +104,30 @@ module.exports = type => {
           groupConnections.push(groupConn)
         })
 
-        return groupConnections
+        // make tags arrays case ignored
+        const tags = groupConnections.map(e=>{
```
Comment (jquense) [inline packages/gatsby/src/schema/build-connection-fields.js]: I don't think this is the best spot for this logic. The helpers here are for generically constructing infered types, and having very specific markdown logic here may lead to stuff getting out of sync or ignored in the future. Ideally you'd want to locate this with the code that produces the data, e.g. the transformer-remark plugin. and make use of gatsby's `extendNode` api to tweak the creation of the tags

**Comment by jquense** [comment]
thanks for jumping in on this! I don't actually think there is an option for case insensitive querying unfortunately, in GraphQL (as far as I know)

File: packages/gatsby/src/schema/build-connection-fields.js (hunk 1 above)
Comment (wangsongiam) [inline packages/gatsby/src/schema/build-connection-fields.js]: Yeah, my code was kinda messy definitely needs improvement. I was thinking to make an option for the `group` in `Graphql` query. something like this
```
    allMarkdownRemark {
      group(field: frontmatter___tags, caseSensitive: no) {
        fieldValue
        totalCount
      }
```

Comment (wangsongiam) [inline packages/gatsby/src/schema/build-connection-fields.js]: Also, by `extendNode`, do you mean `onCreateNode`? Couldn't find it by searching repo. :D

Comment (jquense) [inline packages/gatsby/src/schema/build-connection-fields.js]: sorry this one: i'm still using the old name of hte API https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-remark/src/gatsby-node.js#L2

Comment (jquense) [inline packages/gatsby/src/schema/build-connection-fields.js]: this one tho might illustrate the potential a bit better tthough: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-react-docgen/src/extend-node-type.js

Comment (wangsongiam) [inline packages/gatsby/src/schema/build-connection-fields.js]: Wow, didn't know `gatsby-transformer-react-docgen` exist, not even included on the website,  I'll check on it, thanks!

**Comment by wangsongiam** [comment]
np, thank you for clarifying the potentials

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 2b34b8decf943d6829de3c80166ddc178f653ba8

https://app.netlify.com/sites/using-glamor/deploys/599400517960b10dbf7e43cf

**Comment by KyleAMathews** [comment]
This code sample you added in a comment is more what I was thinking — agree with @jquense that adding specific code for markdown tags isn't going to work since Gatsby works with all sorts of data.

Perhaps close this PR and try again?

```graphql
allMarkdownRemark {
    group(field: frontmatter___tags, caseSensitive: false) {
      fieldValue
      totalCount
    }
}
```

**Pull request closed without merging**